trame_vtk/modules/vtk/addon_serializer.py

```diff
@@ -338,6 +338,88 @@ def cubeAxesSerializer(parent, actor, actorId, context, depth):
     }
 
 
+def imagedataSerializer(
+    parent, dataset, datasetId, context, depth, requested_fields=["Normals", "TCoords"]
+):
+    if hasattr(dataset, "GetDirectionMatrix"):
+        direction = [dataset.GetDirectionMatrix().GetElement(0, i) for i in range(9)]
+    else:
+        direction = [1, 0, 0, 0, 1, 0, 0, 0, 1]
+
+    # Extract dataset fields
+    fields = []
+    extractRequiredFields(fields, parent, dataset, context, "*")
+
+    return {
+        "parent": render_window_serializer.getReferenceId(parent),
+        "id": datasetId,
+        "type": render_window_serializer.class_name(dataset),
+        "properties": {
+            "spacing": dataset.GetSpacing(),
+            "origin": dataset.GetOrigin(),
+            "extent": dataset.GetExtent(),
+            "direction": direction,
+            "fields": fields,
+        },
+    }
+
+
+def genericVolumeMapperSerializer(parent, mapper, mapperId, context, depth):
+    # This kind of mapper requires us to get 2 items: input data and lookup
+    # table
+    dataObject = None
+    dataObjectInstance = None
+    lookupTableInstance = None
+    calls = []
+    dependencies = []
+
+    if hasattr(mapper, "GetInputDataObject"):
+        mapper.GetInputAlgorithm().Update()
+        dataObject = mapper.GetInputDataObject(0, 0)
+    else:
+        render_window_serializer.logger.debug(
+            "This mapper does not have GetInputDataObject method"
+        )
+
+    if dataObject:
+        dataObjectId = "%s-dataset" % mapperId
+        dataObjectInstance = render_window_serializer.serializeInstance(
+            mapper, dataObject, dataObjectId, context, depth + 1
+        )
+
+        if dataObjectInstance:
+            dependencies.append(dataObjectInstance)
+            calls.append(
+                ["setInputData", [render_window_serializer.wrapId(dataObjectId)]]
+            )
+
+    if dataObjectInstance:
+        if hasattr(mapper, "GetImageSampleDistance"):
+            imageSampleDistance = mapper.GetImageSampleDistance()
+        else:
+            imageSampleDistance = 1.0
```

Comment:
I think this is the best we can do? 1.0 seems to be a reasonable default

Reply:
sure